lib/http-bridge/test/integration/Main.hs

```diff
@@ -172,13 +184,16 @@ main = hspec $ do
     -- code coverage measures from running the scenarios on top of it!
     cardanoWalletServer
         :: NetworkLayer (HttpBridge 'Testnet) IO
-        -> Int
-        -> IO ()
-    cardanoWalletServer nl serverPort = void $ forkIO $ do
-        db <- MVar.newDBLayer
-        let tl = HttpBridge.newTransactionLayer
-        wallet <- newWalletLayer nullTracer block0 db nl tl
-        Server.start (const $ pure ()) (Just serverPort) wallet
+        -> Listen
+        -> IO Port
+    cardanoWalletServer nl walletListen = do
+        port <- newEmptyMVar
+        void . forkIO $ do
+            db <- MVar.newDBLayer
+            let tl = HttpBridge.newTransactionLayer
+            wallet <- newWalletLayer nullTracer block0 db nl tl
+            Server.start (putMVar port) walletListen wallet
```

Comment:
That's a good idea and a good way to test that the random port binding works as expected :tada: